test/LANDRegistry.js

```diff
@@ -1098,4 +1099,176 @@ contract('LANDRegistry', accounts => {
+        let deemPeriod = await land.deemPeriod()
+        deemPeriod.should.be.bignumber.equal(0)
+        
+        const date = new Date()
+        const oneMonth = new Date(date.getFullYear(), date.getMonth() + 1, 0)
+          .getDate() * 24 * 60 * 60 // In seconds
+
+        await land.setDeemPeriod(oneMonth, sentByCreator)
+        deemPeriod = await land.deemPeriod()
+
+        deemPeriod.should.be.bignumber.equal(oneMonth)
+      })
+
+      it('reverts if hacker set deem period', async function() {
+        const date = new Date()
+        const oneMonth = new Date(date.getFullYear(), date.getMonth() + 1, 0)
+          .getDate() * 24 * 60 * 60 // In seconds
+
+        await assertRevert(land.setGracePeriod(oneMonth, sentByHacker))
+      })
+
+    })
+
+    describe('hasDecayed', function() {
+      beforeEach(async function() {
+        await land.pingMyself(sentByUser)
+      })
+
+      it('should return true if a LAND is decayed', async function() {
+        const assetId = await land.encodeTokenId(0, 1)
+        const date = new Date()
+        const oneMonth = new Date(date.getFullYear(), date.getMonth() + 1, 0)
+          .getDate() * 24 * 60 * 60 // In seconds
+        
+        // 2 seconds from now
+        await land.setGracePeriod(2, sentByCreator)
+        // 1 second
+        await land.setDeemPeriod(1, sentByCreator)
+        await sleep(4) // inside delta period
+        // One second
+        const decayed = await land.hasDecayed(assetId)
+        decayed.should.be.true
+      })
+
+      it('should return false if not decayed LAND but near to deem period', async function() {
+        const assetId = await land.encodeTokenId(0, 1)
+        const date = new Date()
+        const oneMonth = new Date(date.getFullYear(), date.getMonth() + 1, 0)
+          .getDate() * 24 * 60 * 60 // In seconds
+        
+        // 2 seconds from now
+        await land.setGracePeriod(2, sentByCreator)
+        // One month
+        await land.setDeemPeriod(oneMonth, sentByCreator)
+        const gp = await land.gracePeriod()
+        const dp = await land.deemPeriod()
+        await sleep(3) // inside delta period
+        const decayed = await land.hasDecayed(assetId)
+        decayed.should.be.false
+      })
+
+
+      it('should return false if no gracePeriod set', async function() {
+        const assetId = await land.encodeTokenId(0, 1)
+        const decayed = await land.hasDecayed(assetId)
+        decayed.should.be.false
+      })
+
+      it('should return false if gracePeriod is bigger than today (one month from now)', async function() {
+        const assetId = await land.encodeTokenId(0, 1)
+        const date = new Date()
+        const oneMonth = new Date(date.getFullYear(), date.getMonth() + 1, 0)
+          .getDate() * 24 * 60 * 60 // In seconds
+
+        await land.setGracePeriod(oneMonth, sentByCreator)
+        const decayed = await land.hasDecayed(assetId)
+        decayed.should.be.false
```

Comment:
It could be returning false because the deeming period is not set. Do you mind checking it too?

Reply:
Pending